pylops/utils/decorators.py

```diff
@@ -4,22 +4,22 @@
 
 
 def add_ndarray_support_to_solver(func):
-    """Decorator which converts a solver-type function which only supports
+    """Decorator which converts a solver-type function that only supports
     a 1d-array into one that supports one (dimsd-shaped) ndarray.
 
     Parameters
     ----------
     func : :obj:`callable`
-        Solver type function. Its signature must be ``func(A, b, x0=None, **kwargs)``.
+        Solver type function. Its signature must be ``func(A, b, *args, **kwargs)``.
         Its output must be a result-type tuple: ``(xinv, ...)``.
     """
 
     @wraps(func)
-    def wrapper(A, b, x0=None, **kwargs):  # SciPy-type signature
-        if x0 is not None:
-            x0 = x0.ravel()
+    def wrapper(A, b, *args, **kwargs):  # SciPy-type signature
```

Comment:
I initially though about using this signature. What happens then the scipy solver is called like this `cg_wrapped(A, b, x0)` (x0 is args[0])? I haven't tried it but I think it wouldnt work.

Reply:
Mmh maybe, but the one we have so far can definitely not be used for a couple of solvers which require more mandatory parameters than A and y…

In theory if x0 is a kwarg and someone passes it as arg, it is not surprising that this will raise an error… but then we are basically saying we need x0 as 3rd argument all the time no matter it is passed as x0=x0 or x0? I don’t think we can do that for all solvers, then we may need multiple of these or we need to find a way to write solvers that don’t rely on this?

Reply:
Also for example scipy lsqr does not have x0 as 3rd element so I think calling A,y,x0 is not really something we should accommodate for… what do you think?

Reply:
One other option could be to make a decorator `disabled_ndarray_multiplication` which we can add to solvers like `RegularizedInversion` for the moment as they are not compatible with ndarray multiplications. So that we leave `add_ndarray_support_to_solver ` to pure solvers. What do you think?

Reply:
I would personally keep `add_ndarray_support_to_solver` only for pure solvers. This way works.